source/common/network/ioctl_socket_option_impl.cc

```diff
@@ -0,0 +1,57 @@
+#include "common/network/ioctl_socket_option_impl.h"
+
+#include "envoy/common/exception.h"
+#include "envoy/config/core/v3/base.pb.h"
+
+#include "common/api/os_sys_calls_impl.h"
+#include "common/common/assert.h"
+#include "common/common/scalar_to_byte_vector.h"
+#include "common/common/utility.h"
+#include "common/network/address_impl.h"
+
+namespace Envoy {
+namespace Network {
+
+// Socket::Option
+bool IoctlSocketOptionImpl::setOption(
+    Socket& socket, envoy::config::core::v3::SocketOption::SocketState state) const {
+  if (in_state_ == state) {
+    if (!optname_.hasValue()) {
+      ENVOY_LOG(warn, "Failed to set unsupported control on socket");
+      return false;
+    }
+
+    unsigned long size = 0;
+    const Api::SysCallIntResult result = socket.genericIoctl(
+        optname_.option(), inBuffer_, inBuffer_size_, outBuffer_, outBuffer_size_, size);
+    if (result.rc_ != 0) {
+      ENVOY_LOG(warn, "Setting {} control on socket failed: {}", optname_.name(),
+                errorDetails(result.errno_));
+      return false;
+    }
+  }
+
+  return true;
+}
+
+void IoctlSocketOptionImpl::hashKey(std::vector<uint8_t>& hash) const {
+  std::string in_buffer_bstr(reinterpret_cast<const char*>(inBuffer_), inBuffer_size_);
+  std::string out_buffer_bstr(reinterpret_cast<const char*>(outBuffer_), outBuffer_size_);
+  pushScalarToByteVector(
+      StringUtil::CaseInsensitiveHash()(std::string(in_buffer_bstr + out_buffer_bstr)), hash);
+}
+
+absl::optional<Socket::Option::Details>
+IoctlSocketOptionImpl::getOptionDetails(const Socket&,
+                                        envoy::config::core::v3::SocketOption::SocketState) const {
+
+  std::string in_buffer_bstr(reinterpret_cast<const char*>(inBuffer_), inBuffer_size_);
```

Comment:
naming: `_bstr`: what does that mean?

Reply:
absl::string_view would remove a copy

Reply:
> naming: _bstr: what does that mean?

Binary string. I took the name from other places in the codebase that did a similar cast and kept the suffix for consistency. No strong opinion though, I can change it

Reply:
> absl::string_view would remove a copy

This is a bit tricky because we need to concatenate the strings and this would require a copy since `string_view ` does not support either `insert` or `operator +`. I will think about it more